analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInGenericClass.cs

```diff
@@ -64,15 +66,16 @@ public sealed class StaticFieldInGenericClass : SonarDiagnosticAnalyzer
                         field.Declaration.Variables.ToList().ForEach(variable => CheckMember(variable, variable.Identifier.GetLocation(), typeParameterNames, c));
                     }
 
-                    var properties = classDeclaration.Members
+                    var properties = typeDeclaration.Members
                         .OfType<PropertyDeclarationSyntax>()
                         .Where(p => p.Modifiers.Any(SyntaxKind.StaticKeyword))
                         .ToList();
 
                     properties.ForEach(property => CheckMember(property, property.Identifier.GetLocation(), typeParameterNames, c));
 
                 },
-                SyntaxKind.ClassDeclaration);
+                SyntaxKind.ClassDeclaration,
+                SyntaxKindEx.RecordDeclaration);
 
         private static void CheckMember(SyntaxNode root, Location location, IEnumerable<string> typeParameterNames,
             SyntaxNodeAnalysisContext context)
```

Comment:
pedantic: we can move `SyntaxNodeAnalysisContext context` parameter to line 77

(also the one from line 89)